Fix spack bootstrap without setup-env

Jordan474 opened the pull request:
Allow spack bootstrap to work without having spack in your `PATH` (without `source share/spack/setup-env.sh`). And also without `gmake`.

Removed depfile-generated `Makefile` install mechanism, see https://github.com/spack/spack/pull/41458#issuecomment-1843431707.

Since #34029, spack bootstrapp can fail if spack isn't in your `PATH`:

```sh
$ ./bin/spack bootstrap now --dev
==> [BOOTSTRAPPING] Installing dependencies (py-isort@5, py-mypy@0.900:, py-black@:23.1.0, py-flake8@3.8.2:, py-pytest@6.2.4:)
==> Error: Command exited with status 2:
'/usr/bin/make' '-C' '/home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd' '-j' '16'
```

This is because the generated `Makefile` tries to run `spack`.

I thought of adding an absolute spack path in the generated `Makefile`, but I think having a relocatable `Makefile` can be useful.

Commit:
bootstrap: Fix without setup-env.sh (./bin/spack bootstrap now)

Commit:
Fix style mypy

haampie commented on lib/spack/spack/bootstrap/environment.py:
```diff
@@ -133,11 +134,14 @@ def _install_with_depfile(self) -> None:
             str(self.environment_root()),
             "-j",
             str(spack.util.cpus.determine_number_of_jobs(parallel=True)),
+            extra_env=extra_env,
```
```suggestion
            f"SPACK={spack.paths.spack_script}",
```

haampie replied:
Or maybe even `f"SPACK={sys.executable} {spack.paths.spack_script}"` in case also Python is not in the PATH.

haampie replied:
Actually, see comment below. Let's just drop the makefile.

haampie commented:
> I thought of adding an absolute spack path

I believe the file is always generated on install, so should be fine.

Note that this PR might only survive temporarily, cause I'm removing it again in #40773 because of some other issues...

Commit:
Don't bootstrap with depfile generated Makefile

Commit:
Respect SuppressOutput(msg_enabled=False) during install

Commit:
Suppress bootstrap output (like it was with depfile)

Jordan474 commented:
I removed the depfile Makefile from bootstrap. Updated the PR description.

Without depfile, my `spack boostrap --dev` went from 20s to 30s on a laptop (with clingo binary buildcache).

But, I kind of agree, using depfile for bootstrapping is not ideal. Even if I could not do without depfile in production for large spack env.

Commit:
Fix lint
